OpenRA.Mods.RA2/Orders/EnterGarrisonTargeter.cs

```diff
@@ -0,0 +1,77 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using OpenRA.Mods.Common.Orders;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Mods.RA2.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.RA2.Orders
+{
+    public class EnterGarrisonTargeter<T> : UnitOrderTargeter where T : ITraitInfo
```

Comment:
Why are you using a template within this one? Isn't this only used for Garrison?

Soulnds like a perf overhead for treating it entirely as a generic.

Reply:
changed to public class EnterGarrisonOrderTargeter<GarrisonableInfo> : UnitOrderTargeter in the next commit. That should work

Reply:
Actually that causes problems further down. I'll look into How to fix this.  I'm not sure how to use !target.Info.HasTraitInfo<T>()  later on Lines 59  and 69 without T being defined. 
I'll do some more research into how Generics work.

Reply:
Fixed this.